paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/DefaultConfirmationHandler.kt

```diff
@@ -507,6 +507,25 @@ internal class DefaultConfirmationHandler(
         }
     }
 
+    private fun onResult(result: ConfirmationDefinition.Result) {
+        val confirmationResult = when (result) {
+            is ConfirmationDefinition.Result.Succeeded -> ConfirmationHandler.Result.Succeeded(
+                intent = result.intent,
+                deferredIntentConfirmationType = result.deferredIntentConfirmationType,
+            )
+            is ConfirmationDefinition.Result.Failed -> ConfirmationHandler.Result.Failed(
+                cause = result.cause,
+                type = result.type,
+                message = result.message,
+            )
+            is ConfirmationDefinition.Result.Canceled -> ConfirmationHandler.Result.Canceled(
+                action = result.action,
+            )
+        }
+
+        onIntentResult(confirmationResult)
+    }
```

Comment:
Feels weird here using `ConfirmationDefinition.Result` directly here while we have a separate type for `Action` in `ConfirmationMediator`. Unfortunately, there isn't a use case to make a separate mediator result type. `Action` needed a type internal to the `ConfirmationMediator` and it translated the generic action returned from the definition to a non-generic action. `ConfirmationDefinition.Result` is already non-generic.